ckanext/apis/logic/auth.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import ckan.model as model
+from ckan.plugins import toolkit
+
+log = logging.getLogger(__name__)
+
+
+def get_auth_functions():
+    return {
+        'apiset_create': apiset_create,
+        'apiset_package_association_create': apiset_association_create,
+        'apiset_delete': apiset_delete,
+        'apiset_package_association_delete': apiset_association_delete,
+        'apiset_update': apiset_update,
+        'apiset_admin_remove': remove_apiset_admin,
+        'apiset_package_list': apiset_package_list,
+        'package_apiset_list': package_apiset_list,
+        'apiset_admin_list': apiset_admin_list,
+    }
+
+
+@toolkit.auth_allow_anonymous_access
+def apiset_package_list(context, data_dict):
+    return {'success': True}
+
+@toolkit.auth_allow_anonymous_access
+def package_apiset_list(context, data_dict):
+    return {'success': True}
+
+def apiset_admin_list(context, data_dict):
+    return {'success': False}
+
+def remove_apiset_admin(context, data_dict):
+    return {'success': False}
+
+def apiset_update(context, data_dict):
+    user = context.get('user')
+    try:
+        toolkit.check_access('package_update', context, data_dict)
+        return {'success': True}
+    except toolkit.NotAuthorized:
+        return {'success': False,
+                'msg': toolkit._('User {0} not authorized to edit apisets').format(user)}
+
+def apiset_create(context, data_dict):
+    user = context.get('user')
+    try:
+        toolkit.check_access('package_create', context, data_dict)
+        return {'success': True}
+    except toolkit.NotAuthorized:
+        return {'success': False,
+                'msg': toolkit._('User {0} not authorized to create apisets').format(user)}
+
+def apiset_association_create(context, data_dict):
+    user = context.get('user')
+    try:
+        toolkit.check_access('package_create', context, data_dict)
```

Comment:
Should this actually check update access (I think that associating something for apiset is more like an update than create)

Reply:
Implemented in [48e44f1](https://github.com/vrk-kpa/ckanext-apis/pull/7/commits/48e44f1abb34844a2b5579e0de5e99e033c81dd7)